src/bytecode.rs

```diff
@@ -78,5 +83,18 @@ pub fn to_bytecode(ast_node: AstNode, prog: &mut Vec<Instruction>) {
             prog.push(Instruction::Assign { id })
         }
         AstNode::ID { value } => prog.push(Instruction::Load { id: value }),
+        AstNode::Boolean { value } => prog.push(Instruction::Push {
+            value: StackValue::Boolean(value),
+        }),
+        AstNode::GreaterThan { lhs, rhs } => {
```

Comment:
Have you considered grouping the operators in a nested enum? (e.g. `AstNode::BinaryOp(BinaryOp)` where `enum BinaryOp { LessThan, GreaterThan }`?

If you ever need to match in a context where you expect only binary operators, it means you can take advantage of Rust's exhaustive match detection.

Reply:
Will it make sense to keep AST nodes as is: `AstNode::GreaterThan` and `AstNode::LessThan` and group enums for bytecodes Instruction::BinaryOp?

Reply:
done :point_right: 8de878c85feeb9c126d948c63bdb6711483586c7